lib/Agrammon/Web/Service.pm6

```diff
@@ -84,8 +119,15 @@ class Agrammon::Web::Service {
         return $newUser;
     }
 
-    method change-password(Agrammon::Web::SessionUser $user, Str $old, Str $new) {
-        $user.change-password($old, $new);
+    method change-password(Agrammon::Web::SessionUser $user, Str $oldPassword, Str $newPassword) {
+        ...
+        $user.change-password($oldPassword, $newPassword);
+        return $user;
+    }
+
+    method reset-password(Agrammon::Web::SessionUser $user, Str $email, Str $password, Str $key) {
+        ...
+        $user.reset-password($email, $password);
         return $user;
     }
```

Comment:
Here I see it *does* expect to be passed the user. So I guess the route code needs updating.

Reply:
Ah, now I see the "Parameters might not all be correct yet" mention at the top of the PR, maybe it's about this.

Reply:
See above.

Reply:
> Ah, now I see the "Parameters might not all be correct yet" mention at the top of the PR, maybe it's about this.

Actually I meant that I have to bring the parameters the frontend sends in sync with what the backend expects. This is probably correct for most routes, but I will have to verify that and in some cases the parameters might be a bit strangely named.

However, I found it easier to make the backend somewhat consistent and add route tests for it first, then in the next PR add the missing webservice calls (with tests) and then in the third step bring the frontend in sync.